clients/drcachesim/tools/filter/type_filter.h

```diff
@@ -78,13 +78,33 @@ class type_filter_t : public record_filter_t::record_filter_func_t {
         }
     }
     void *
-    parallel_shard_init(memtrace_stream_t *shard_stream) override
+    parallel_shard_init(memtrace_stream_t *shard_stream,
+                        bool partial_trace_filter) override
     {
-        return nullptr;
+        per_shard_t *per_shard = new per_shard_t;
+        per_shard->partial_trace_filter = partial_trace_filter;
+        return per_shard;
     }
     bool
-    parallel_shard_filter(const trace_entry_t &entry, void *shard_data) override
+    parallel_shard_filter(trace_entry_t &entry, void *shard_data) override
     {
+        per_shard_t *per_shard = reinterpret_cast<per_shard_t *>(shard_data);
+        if (entry.type == TRACE_TYPE_MARKER && entry.size == TRACE_MARKER_TYPE_FILETYPE) {
+            if (TESTANY(entry.addr, OFFLINE_FILE_TYPE_ENCODINGS) &&
+                !per_shard->partial_trace_filter &&
```

Comment:
Not sure I understand this: if we filter the first half of a trace and remove all the encodings, don't we still want to remove the encoding flag from the type?  The partial trace will be output as a whole new trace and its type should be the same as when we filter the entire trace, right?

Reply:
> if we filter the first half of a trace and remove all the encodings, don't we still want to remove the encoding flag from the type?

Since there still are encodings present in the trace, I thought we should keep the OFFLINE_FILE_TYPE_ENCODINGS bit.

> The partial trace will be output as a whole new trace

No the partial trace is still a part of the overall trace, with the second half still having encodings.